agreement_legal_sale_fieldservice/models/sale_order.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _action_confirm(self):
+        res = super(SaleOrder, self)._action_confirm()
+        for order in self:
+            agreement_id = self.env['agreement'].\
+                search([('code', '=', order.name)])
```

Comment:
Not sure, but isn't there a sale_id field in the Agreements model, that can be used for this?